src/main/java/backend/Logic.java

```diff
@@ -85,22 +85,26 @@ public void refresh() {
     }
 
     public CompletableFuture<Boolean> openPrimaryRepository(String repoId) {
-        return openRepository(repoId, true);
+        assert Utility.isWellFormedRepoId(repoId);
+        prefs.setLastViewedRepository(repoId);
+        if ((isAlreadyOpen(repoId) && repoId.equals(getDefaultRepo())) || models.isRepositoryPending(repoId)) {
```

Comment:
to add on to the above, could you provide a list of steps that can reproduce the issue that the extra check is meant to solve? (list of steps applied to the current master should demonstrate the issue, same list of steps applied to your branch should not demonstrate the issue)